server/provider/networkProvider.go

```diff
@@ -214,6 +226,11 @@ func (provider *networkProvider) doGetBlockByNonce(nonce uint64) (*data.Block, e
 		WithLogs:         true,
 	}
 
+	block, ok := provider.getBlockByNonceCached(nonce)
```

Comment:
what if we are talking about real-time data and there is a rollback?

Reply:
Good question! In rosetta, we only consider blocks under final:

https://github.com/ElrondNetwork/rosetta/blob/main/server/provider/nodeStatus.go#L60